services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/config/JdbcProperties.java

```diff
@@ -115,28 +116,20 @@ public void setTableName(final String tableName) {
     public static JDBCClient dataSource(final Vertx vertx, final JdbcProperties dataSourceProperties) {
 
         final JsonObject config = new JsonObject()
-                .put("url", dataSourceProperties.getUrl())
-                .put("user", dataSourceProperties.getUsername());
-
-        // password is added later, after logging
-
-        if (dataSourceProperties.getDriverClass() != null) {
-            config.put("driver_class", dataSourceProperties.getDriverClass());
-        }
-        if (dataSourceProperties.getMaximumPoolSize() != null) {
-            config.put("max_pool_size", dataSourceProperties.getMaximumPoolSize());
-        }
+                .put("dataSourceImplementation", "AGROAL")
+                .put("jdbcUrl", dataSourceProperties.getUrl())
+                .put("principal", dataSourceProperties.getUsername())
+                .put("maxSize", dataSourceProperties.getMaximumPoolSize());
```

Comment:
Quarkus Agroal related property names can be seen here; https://github.com/vert-x3/vertx-jdbc-client/blob/ccaa53f4361ef942891a40a1f4e9abfa92fef575/src/main/java/io/vertx/ext/jdbc/spi/impl/AgroalCPDataSourceProvider.java#L59 - as a reference for PR [#3548] too.